src/explorers/blockchain-explorer.js

```diff
@@ -0,0 +1,63 @@
+import { SUB_STEPS, TRANSACTION_APIS, TRANSACTION_ID_PLACEHOLDER } from '../constants';
+import { buildTransactionApiUrl } from '../services/transaction-apis';
+import { request } from '../services/request';
+import { VerifierError } from '../models';
+import { getText } from '../domain/i18n/useCases';
+import { generateTransactionDataFromBitpayResponse } from './bitcoin/bitpay';
+import { generateTransactionDataFromBlockcypherResponse } from './bitcoin/blockcypher';
+import { generateTransactionDataFromBlockexplorerResponse } from './bitcoin/blockexplorer';
+import { generateTransactionDataFromBlockstreamResponse } from './bitcoin/blockstream';
+import { generateTransactionDataFromEtherscanResponse } from './ethereum/etherscan';
+import { isTestChain } from '../constants/blockchains';
+import { generateTransactionDataFromBlockcypherEthereumResponse } from './ethereum/blockcypher-ethereum';
+
+function isTransactionGeneratorAsync (chain) {
+  return chain === TRANSACTION_APIS.Etherscan;
+}
+
+export async function getBlockchainTransactionFromApi (apiName, transactionId, chain) {
+  const requestUrl = buildTransactionApiUrl({
+    apiName,
+    searchValue: TRANSACTION_ID_PLACEHOLDER,
+    newValue: transactionId,
+    testApi: isTestChain(chain)
+  });
+
+  return new Promise((resolve, reject) => {
+    return request({ url: requestUrl }).then(response => {
+      try {
```

Comment:
maybe make this `try/catch` part of another function? This bit is a bit crazy to follow.